File: src/main/java/com/dbserver/ugo/votacao/pauta/PautaService.java
package com.dbserver.ugo.votacao.pauta;

import com.dbserver.ugo.votacao.pauta.exception.PautaNotFoundException;
import jakarta.persistence.EntityNotFoundException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class PautaService {

    private final PautaRepository pautaRepository;
    private final PautaMapper pautaMapper;


    public PautaResponseDTO criar(PautaCreateDTO dto) {
        Pauta pauta = pautaMapper.toEntity(dto);
        Pauta saved = pautaRepository.save(pauta);
        return pautaMapper.toDTO(saved);
    }


    public PautaResponseDTO atualizar(Long idPauta, PautaPutDTO dto) {
        Pauta entity = pautaRepository.findById(idPauta)
                .orElseThrow(() -> new PautaNotFoundException(idPauta));
        pautaMapper.updateFromPut(dto, entity);
        return pautaMapper.toDTO(pautaRepository.save(entity));
    }



    public PautaResponseDTO atualizarParcial(Long idPauta, PautaPatchDTO dto) {
        Pauta entity = pautaRepository.findById(idPauta)
                .orElseThrow(() -> new PautaNotFoundException(idPauta));
        pautaMapper.updateFromPatch(dto, entity);
        pautaRepository.save(entity);
        return pautaMapper.toDTO(entity);
    }


    public PautaResponseDTO buscarPorId(Long idPauta) {
        Pauta entity = pautaRepository.findById(idPauta)
                .orElseThrow(() -> new PautaNotFoundException(idPauta));
        return pautaMapper.toDTO(entity);
    }

    public List<PautaResponseDTO> listar() {
        return pautaRepository.findAll()
                .stream()
                .map(pautaMapper::toDTO)
                .toList();
    }

    public void deletar(Long idPauta) {
        if (!pautaRepository.existsById(idPauta)) {
            throw new PautaNotFoundException(idPauta);
        }
        pautaRepository.deleteById(idPauta);
    }

}

File: frontend/src/services/pautaService.ts
import { http } from "../api/httpClient";
import type {
  PautaCreateDTO,
  PautaPutDTO,
  PautaPatchDTO,
  PautaResponseDTO
} from "../types/pauta";

export const pautaApi = {
  async listar(): Promise<PautaResponseDTO[]> {
    const res = await http.get<PautaResponseDTO[]>("/pautas");
    return res.data;
  },

  async buscarPorId(id: number): Promise<PautaResponseDTO> {
    const res = await http.get<PautaResponseDTO>(`/pautas/${id}`);
    return res.data;
  },

  async criar(data: PautaCreateDTO): Promise<PautaResponseDTO> {
    const res = await http.post<PautaResponseDTO>("/pautas", data);
    return res.data;
  },

  async atualizar(id: number, data: PautaPutDTO): Promise<PautaResponseDTO> {
    const res = await http.put<PautaResponseDTO>(`/pautas/${id}`, data);
    return res.data;
  },

  async atualizarParcial(id: number, data: PautaPatchDTO): Promise<PautaResponseDTO> {
    const res = await http.patch<PautaResponseDTO>(`/pautas/${id}`, data);
    return res.data;
  },

  async deletar(id: number): Promise<void> {
    await http.delete(`/pautas/${id}`);
  }
};
